crates/core/tedge/src/cli/connect/common_mosquitto_config.rs

```diff
@@ -135,15 +136,15 @@ impl CommonMosquittoConfig {
     pub fn with_external_opts(
         self,
         port: Option<u16>,
-        bind_address: Option<String>,
+        bind_address: Option<IpAddress>,
```

Comment:
I think that may be incorrect because hosts can be also DNS names, but I'm not sure @didier-wenzek

Reply:
For this specific case, I think `IpAddress` is more appropriate. A DNS name makes sense for `mqtt.client.host` but not the `mqtt.external.bind_address` that is the address bound by the local listener.